### DISCUSSION TITLE
Misunderstanding poetry resolving system version resolution

### DISCUSSION THREAD

**Discussion by vit1251** [python-poetry/poetry#5784] [Q&A] [answered]
Hello, I use poetry with large dependency graph  (~ 30 dependencies on top level)

Specify dependencies:

```
python = ">=3.7"
pandas = "1.3"
```

Resolve produce conflict:

```
 fact: pandas (1.3.0) requires Python >=3.7.1
   1: conflict: pandas (1.3.0) requires Python >=3.7.1
   1: !  pandas (1.3.0) is satisfied by  pandas (1.3)
   1: ! which is caused by "test (2) depends on pandas (1.3)"
   1: ! thus: version solving failed
   1: Version solving took 261.921 seconds.
   1: Tried 1 solutions.
```

I use Python 3.7.3 and Poetry version 1.1.13 unable to resolve dependencies. 
Could you please explain conflict root cause?

**Comment by TillerBurr** [comment, marked as answer]
With those dependencies, it allows for the possibility of `python==3.7.0`, which is a conflict for `pandas==1.3.0`.  I would try specifying 
```
python = ">=3.7.1"
```

**Reply by vit1251** [reply]
Why poetry does not seek Python version up to 3.7.1 himself (it allowed by rule >=3.7) instead and cause an error?

**Reply by TillerBurr** [reply]
Not sure what you mean. 

The dependency resolution doesn't take into account python versions currently installed. If it is set to `python=">=3.7"`, then it's going to assume that anyone with python 3.7.0 can install the package provided by pyproject.toml.

It's when you try to install the package that the installed version is looked at. If I had python 3.6 installed and tried to install this, then you would get an error about not using the correct python version.

**Reply by vit1251** [reply]
Hm. Now I seem to understand what was the matter. Thanks.
